cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -497,9 +498,6 @@ TEST(GcsFileSystem, FileSystemCompare) {
   b_options.project_id = "test-only-invalid-project-id";
   auto b = GcsFileSystem::Make(b_options);
   EXPECT_THAT(b, NotNull());
-  EXPECT_TRUE(b->Equals(*b));
-
-  EXPECT_FALSE(a->Equals(*b));
```

Comment:
Can we revert these changes?

Reply:
Reverted! Ref e5b606e462cb362ef49092eb10797d166e2e60fd.